### DISCUSSION TITLE
Playing mkv format videos will crash every few minutes.

### DISCUSSION THREAD

**Discussion by DreamMaoMao** [smplayer-dev/smplayer#785] [General]
![image](https://github.com/smplayer-dev/smplayer/assets/30348075/75feaaa9-3301-4308-ac61-9b207d097875)
![image](https://github.com/smplayer-dev/smplayer/assets/30348075/9f132bb5-bdf6-4dc2-80b2-703f4da3246a)
![image](https://github.com/smplayer-dev/smplayer/assets/30348075/b0151ea3-fe3e-4c78-aa88-ef985f647f0e)

**Comment by DreamMaoMao** [comment]
when disable hardware decode , it will not crash again. what's wrong about that, i just want to use hardware decode.

**Reply by Raza-0007** [reply]
Most likely vaapi hwdec is not stable on your system. Put hwdec in "auto" and let smplayer/mpv find the best hwdec that works on your system and then stick to that one.

Other than that I cannot say what the issue is since you did not post any logs.

**Reply by DreamMaoMao** [reply]
I have been observing the logs of smplayer and mpv while using it, and there was no abnormal output in the logs after the crash. However, after I changed my video source from 4k to 1080p, this problem seems to no longer occur. May it be a performance problem under high load

**Reply by Raza-0007** [reply]
> May it be a performance problem under high load

That could be the issue if you have a low end or an integrated graphics card, but in most other cases, offloading video decoding to GPU actually helps with performance, because your GPU is practically sitting idle during video playback, and if you offload decoding to GPU, you actually improvement system performance, as your CPU resources that were being consumed by decoding are now available.

Also .mkv is a container format and has nothing to do with hwdec. Hwdec is used in decoding the codecs that were used to encode the video, like H264, H265 etc.  

I would recommend you read the mpv's [hwdec user guide](https://mpv.io/manual/master/#options-hwdec) for anything that applies to your system. For example, on vaapi it says

>The vaapi mode, if used with --vo=gpu, requires Mesa 11, and most likely works with Intel and AMD GPUs only. It also requires the opengl EGL backend.

I do not know if the above applies to you or not, but it may be a factor.

~Raza
